python/ray/dag/compiled_dag_node.py

```diff
@@ -386,6 +386,118 @@ def __init__(
             assert not isinstance(val, ChannelInterface)
             assert not isinstance(val, DAGInputAdapter)
 
+        # Input reader to read input data from upstream DAG nodes.
+        self.input_reader: ReaderInterface = SynchronousReader(self.input_channels)
+        # Output writer to write output data to downstream DAG nodes.
+        self.output_writer: WriterInterface = SynchronousWriter(self.output_channel)
+        # Store the intermediate result of a READ or COMPUTE operation.
+        # The result of a READ operation will be used by a COMPUTE operation,
+        # and the result of a COMPUTE operation will be used by a WRITE operation.
+        self._intermediate_buffer: Any = None
+
+    def cancel(self):
+        self.input_reader.close()
+        self.output_writer.close()
+
+    def prepare(self):
+        """
+        Prepare the task for execution. The `exec_operation` function can only
```

Comment:
can you define "prepare" more clearly? E.g., "this API has to be used before using `exec_operation`" something like that

Reply:
"this API has to be used before using exec_operation" seems to be the same as the existing comment.

```
        Prepare the task for execution. The `exec_operation` function can only
        be called after `prepare` has been called.
```